docs/guide/ethereum-provider.md

```diff
@@ -112,83 +173,359 @@ params: [
+
+To disable this behavior, set this property to `false` immediately after detecting the provider:
 
 ```javascript
-ethereum.on('accountsChanged', function (accounts) {
-  // Time to reload your interface with accounts[0]!
-});
+ethereum.autoRefreshOnNetworkChange = false;
+```
+
+### ethereum.networkVersion (DEPRECATED)
+
+::: warning
+You should always prefer the chain ID over the network ID.
+
+If you must get the network ID, use [`ethereum.request({ method: 'net_version' })`](#ethereum-request-args).
+
+The value of this property can change at any time.
+:::
+
+A decimal string representing the current blockchain's network ID.
+
+### ethereum.selectedAddress (DEPRECATED)
+
+::: warning
+Use [`ethereum.request({ method: 'eth_accounts' })`](#ethereum-request-args) instead.
+
+The value of this property can change at any time.
+:::
+
+Returns a hexadecimal string representing the user's "currently selected" address.
+
+The "currently selected" address is the first item in the array returned by `eth_accounts`.
+
+## Legacy Methods
+
+### ethereum.enable() (DEPRECATED)
+
+::: warning
+Use [`ethereum.request({ method: 'eth_requestAccounts' })`](#ethereum-request-args) instead.
+:::
+
+Alias for `ethereum.request({ method: 'eth_requestAccounts' })`.
+
+### ethereum.sendAsync() (DEPRECATED)
+
+::: warning
+Use [`ethereum.request()`](#ethereum-request-args) instead.
+:::
+
+```typescript
+interface JsonRpcRequest {
+  id: string | undefined;
+  jsonrpc: '2.0';
+  method: string;
+  params?: Array<any>;
+}
+
+interface JsonRpcResponse {
+  id: string | undefined;
+  jsonrpc: '2.0';
+  method: string;
+  result?: unknown;
+  error?: Error;
+}
+
+type JsonRpcCallback = (error: Error, response: JsonRpcResponse) => unknown;
+
+ethereum.sendAsync(payload: JsonRpcRequest, callback: JsonRpcCallback): void;
 ```
 
-**Note:** The `networkChanged` event is only useful if you disable auto-refresh on network
-change by setting [`ethereum.autoRefreshOnNetworkChange`](#ethereum-autorefreshonnetworkchange) to `false`.
-Otherwise, MetaMask will default to auto-reloading pages upon network change if they have made requests to the provider.
+This is the ancestor of `ethereum.request`. It only works for JSON-RPC methods, and takes a JSON-RPC request payload object and an error-first callback function as its arguments.
+
+See [the Ethereum JSON-RPC API](https://eips.ethereum.org/EIPS/eip-1474) for details.
+
+### ethereum.send() (DEPRECATED)
+
+::: warning
+Use [`ethereum.request()`](#ethereum-request-args) instead.
+:::
+
+```typescript
+ethereum.send(
+  methodOrPayload: string | JsonRpcRequest,
+  paramsOrCallback: Array<unknown> | JsonRpcCallback,
+): Promise<JsonRpcResponse> | unknown;
+```
+
+This method is mostly a Promise-based `sendAsync` with `method` and `params` instead of a payload as arguments.
+However, it behaves in unexpected ways and should be avoided at all costs.
```

Comment:
We should probably be a little more specific about how it behaves, embarrassing as it may be. Some poor soul debugging a dapp might thank us for it someday.

Reply:
GitHub add " 😭" to your reactions you cowards.

Reply:
8f424f3